### ISSUE TITLE
[analyzer] For augmented libraries is it possible to report error where it textually occurs?

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#55448]
The test https://github.com/dart-lang/co19/blob/master/LanguageFeatures/Augmentation-libraries/augmenting_types_A07_t09.dart has this issue. Shortly.
```dart
// augmenting_types_A07_t09.dart

// SharedOptions=--enable-experiment=macros

import augment 'augmenting_types_A07_t09_lib.dart';

abstract interface class I {
  String foo();
}

mixin M {
  int foo() => 42;
}

abstract class C2 with M {} // Error is reported here

main() {
  print(C2);
}

// augmenting_types_A07_t09_lib.dart

// SharedOptions=--enable-experiment=macros

augment library 'augmenting_types_A07_t09.dart';

augment abstract class C2 implements I {} // No expected error here
//                     ^^
// [analyzer] unspecified
// [cfe] unspecified
```
If possible I'd be better to report error where it occurs textually.

cc @eernstg

**Comment by eernstg** [comment]
I would expect that this program should get a treatment which is equivalent to the treatment given to the merged result. After augmentation merging' `augmenting_types_A07_t09.dart` would have the following contents (omitting comments):

```dart
abstract interface class I {
  String foo();
}

mixin M {
  int foo() => 42;
}

abstract class C2 with M implements I {}

main() {
  print(C2);
}
```

For the class `C2`, the computation of a combined member signature for `foo` fails, and `C2` does not override `foo`, and hence we get a compile-time error.

In the merged program we get that error at the name `C2` in the declaration of `C2`. This is probably what we want: An error which is caused by `C2` having the two superinterfaces `M` and `I` simultaneously and thus bringing together two member signatures is an error of `C2`, not an error of `M` or `I` or any declaration of `foo`.

If I understand correctly then the analyzer also reports this error on a declaring occurrence of the name `C2`, and even on the augmenting declaration that causes the declaration of `C2` to switch over from being OK to being an error.

So I'm not sure where it would be, but is there a better textual location where this error could be reported?

**Comment by sgrekhov** [comment]
> If I understand correctly then the analyzer also reports this error on a declaring occurrence of the name C2

Yes, it does. But error occurs after augmentation adds `implements I` to class `C2`. IMHO I'd be nice to report an error in an augmentation library. `abstract class C2 with M {}` itself has no error, problems begin after applying of the augmented library.

**Comment by lrhn** [comment]
We probably do want some "valid interface" checking at each augmentation.

After an augmentation has been applied, the result must have a consistent valid interface. It may not have implementations for everything, that'll be checked at the end, but the `augment C2 implements I` introduces a super-interface which doesn't combine with the existing super-interface, making it impossible to say what the signature of `foo` is for `C2` after that augmentation. That should be an error, and preferably at that point.
(If the same augmentation introduced `Never foo() => throw "ever";`, there would be no problem, because `C2` would have that signature. But it must have *some* valid signature for each member after each augmentation, even if it means recalculating the combined super-interface signatures each time a new interface is added.)
